### PR TITLE
Payment Allocations Two Way Sync - Latest Bugfixes

### PR DESCRIPTION
**Pull request by jbrandolinisf**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by jbrandolinisf** [comment]
@weizsforg - Not sure if Pedro mentioned it, but this is the cleaned up 2 way sync bug fixes to be code reviewed.  This was passed to you given Pedro's bandwidth, and the issues/delays from some branch corruption on github that are now corrected.

**Comment by weizsforg** [comment]
@jbrandolinisf Is there any specific bugs I need to test here or you prefer QA to do the test when we pass to them?

**Comment by jbrandolinisf** [comment]
@weizsforg - If you can look it over from the perspective of did I violate coding standards/do anything silly that way, it would be great.  It probably makes more sense to have QA test the bugs, though you are tagged on them as well in AA.

File: src/classes/ALLO_PaymentSync_SVC.cls [hunk 1]
```diff
@@ -66,13 +72,28 @@ public class ALLO_PaymentSync_SVC {
         Decimal totalPaidUnpaidPayments = 0;
     }
 
+    private void addAllocationAmountToPayment(Id paymentId, Decimal allocAmount) {
```
Comment (weizsforg) [inline src/classes/ALLO_PaymentSync_SVC.cls]: A minor thing. Can we add the Description part of the new methods just as other new methods?

File: src/classes/ALLO_PaymentSync_SVC.cls [hunk 2]
```diff
@@ -66,13 +72,28 @@ public class ALLO_PaymentSync_SVC {
...
+        Decimal amt = allocatedAmountByPayment.containsKey(paymentId) ? allocatedAmountByPayment.get(paymentId) : 0;
+        amt += ((allocAmount != null) ? allocAmount : 0);
+        allocatedAmountByPayment.put (paymentId, amt);
+    }
+
+    private void removeAllocationAmountFromPayment(Id paymentId, Decimal allocAmount) {
```
Comment (weizsforg) [inline src/classes/ALLO_PaymentSync_SVC.cls]: same as above

File: src/classes/ALLO_PaymentSync_SVC.cls [hunk 3]
```diff
@@ -218,6 +233,114 @@ public class ALLO_PaymentSync_SVC {
...
     }
 
+    /*******************************************************************************************************
+    * @description Given a set of Payment Ids, and a set of Opportunity Ids, this method
+    * returns the Payments associated with them, as well as any associated Allocations.
+    * @param Set<Id> paymentIds A set of Payment Ids
+    * @param Set<Id> opportunityIds A set of Opportunity Ids
+    * @return Map<Id, npe01__OppPayment__c> map of Payments with Payment Allocations by Payment Id
+    ********************************************************************************************************/
+    @testVisible Map<Id, npe01__OppPayment__c> retrievePaymentsWithAllocations(Set<Id> paymentIds, Set<Id> opportunityIds) {
```
Comment (weizsforg) [inline src/classes/ALLO_PaymentSync_SVC.cls]: Although the default method modifier in Apex is private. Should we indicate here to make it more clear?

File: src/classes/ALLO_PaymentSync_SVC.cls [hunk 4]
```diff
@@ -218,6 +233,114 @@ public class ALLO_PaymentSync_SVC {
...
+        return pmts;
+    }
+
+    /*******************************************************************************************************
+    * @description Given a set of a set of Opportunity Ids, this method returns the Opportunities
+    * associated with them, as well as any associated Allocations.
+    * @param Set<Id> opportunityIds A set of Opportunity Ids
+    * @return Map<Id, npe01__OppPayment__c> map of Opportunities with Opportunity Allocations by Opportunity Id
+    ********************************************************************************************************/
+    @testVisible Map<Id, Opportunity> retrieveOpportunitiesWithAllocations(Set<Id> opportunityIds) {
```
Comment (weizsforg) [inline src/classes/ALLO_PaymentSync_SVC.cls]: same as the previous method.

**Pull request merged**